discojs/discojs-core/src/models/gpt/index.ts

```diff
@@ -117,13 +87,32 @@ export class GPT extends Model {
     return generatedWords
   }
 
-  static deserialize (weights: WeightsContainer): Model {
-    const model = new GPT()
-    model.weights = weights
+  get config (): Required<GPTConfig> {
+    return this.model.getGPTConfig
+  }
+  override get weights (): WeightsContainer {
+    return new WeightsContainer(this.model.weights.map((w) => w.read()))
+  }
+
+  override set weights (ws: WeightsContainer) {
+    this.model.setWeights(ws.weights)
+  }
+
+  static deserialize (data: GPTSerialization): Model {
+    const model = new GPT(data.config)
+    model.weights = data.weights
     return model
   }
 
-  serialize (): WeightsContainer {
-    return this.weights
+  serialize (): GPTSerialization {
+    return {
+      weights: this.weights,
+      config: this.config
+    }
   }
 }
+
+export type GPTSerialization = {
+  weights: WeightsContainer
+  config?: GPTConfig
+}
```

Comment:
no need to export it, it don't see it being used elsewhere. or maybe you wanted to use it `serialization.model`?